proof/invariant-abstract/DetSchedInvs_AI.thy

```diff
@@ -581,26 +581,30 @@ end
 
 \<comment> \<open>Project threads from the kernel heap\<close>
 
-definition tcb_of :: "kernel_object \<rightharpoonup> tcb" where
-  "tcb_of ko \<equiv> case ko of TCB tcb \<Rightarrow> Some tcb | _ \<Rightarrow> None"
-
-lemma tcb_of_simps:
+primrec tcb_of :: "Structures_A.kernel_object \<rightharpoonup> tcb" where
```

Comment:
In general inside `_AI` theories, we should leave off qualifiers for ASpec, i.e. no `Structures_A.`. ASpec should be the default name space in AInvs, and if it is not for some reason, things should break as immediately as possible.

Reply:
Yes, I should have removed these qualifiers. These definitions are initially used while working in Refine, and that’s where they are coming from.